plugins/modules/zos_volume_init.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) IBM Corporation 2022
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: zos_volume_init
+short_description: Initialize volumes or minidisks on z/OS through ICKDSF init command.
+description:
+  - Wraps the ICKDSF INIT command to initialize volume or minidisk.
+  - Writes volume label and VTOC on volume or minidisk.
+version_added: 1.6.0
+author:
+  - "Austen Stewart (@stewartad)"
+  - "Almigdad Suliman (@Almigdad-Suliman)"
+  - "Nicholas Teves (@TODO)"
+  - "Nuoya Xie (@nxie13)"
+  - "Trevor Glassey (@tkglassey)"
+  - "Tyler Edwards (@TLEdwards-Git)"
+  - "Ketan Kelkar (@ketankelkar)"
+
+options:
+  volume_address:
+    description:
+      - 3 or 4 hexadecimal digit address of the volume to initialize.
+      - This could be a unit address or device number.
+    required: true
+    type: str
+  verify_existing_volid:
```

Comment:
I'd considered having subopts for "verify" but I didnt like that we had mixed types:
```
verify:
  volid: "DEMO12"     <-- string
  offline: yes        <-- bool
  no_data_sets: yes   <-- bool
```

and the following would break or check if current volid is "no" depending on how no was passed (no vs "no")
```
verify:
  volid: no
  offline: yes
  no_data_sets: yes
```

so to skip volid check, the user would have to exclude `volid`:
```
verify:
  offline: yes
  no_data_sets: yes
```
but to skip any others, the user could simply toggle on/off:
```
verify:
  volid: "DEMO12"
  offline: no
  no_data_sets: no
```

Overall, I do like the idea of having a "verify" or "checks" option with suboptions with all available checks -- if i am putting too much emphasis on the mixed types (as demonstrated above) i'd be happy to make this modification

Reply:
I am ok with what you have now in the module, they are described well enough, but to answer your question, i don't think the mixed types is an issue because each sub parameter will have a:
type: str | bool | int etc
default: 
etc

So each sub type is really just another type, ill leave it up to you , feel free to resolve this.